dot/rpc/modules/dev_test.go

```diff
@@ -1 +1,307 @@
+	mockBlockProducerAPI.On("Pause").Return(nil)
+
+	mockErrorBlockProducerAPI.On("Resume").Return(errors.New("babe resume error"))
+	mockBlockProducerAPI.On("Resume").Return(nil)
+
+	mockErrorNetworkAPI.On("Stop").Return(errors.New("network stop error"))
+	mockNetworkAPI.On("Stop").Return(nil)
+
+	mockErrorNetworkAPI.On("Start").Return(errors.New("network start error"))
+	mockNetworkAPI.On("Start").Return(nil)
+
+	type fields struct {
+		networkAPI       NetworkAPI
+		blockProducerAPI BlockProducerAPI
+	}
+	type args struct {
+		r   *http.Request
+		req *[]string
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		args   args
+		expErr error
+		exp    string
+	}{
+		{
+			name: "Not a BlockProducer",
+			fields: fields{
+				nil,
+				nil,
+			},
+			args: args{
+				req: &[]string{"babe", "stop"},
+			},
+			expErr: errors.New("not a block producer"),
+		},
+		{
+			name: "Babe Stop Error",
+			fields: fields{
+				mockNetworkAPI,
+				mockErrorBlockProducerAPI,
+			},
+			args: args{
+				req: &[]string{"babe", "stop"},
+			},
+			exp:    "babe service stopped",
+			expErr: errors.New("babe pause error"),
+		},
+		{
+			name: "Babe Stop OK",
+			fields: fields{
+				mockNetworkAPI,
+				mockBlockProducerAPI,
+			},
+			args: args{
+				req: &[]string{"babe", "stop"},
+			},
+			exp: "babe service stopped",
+		},
+		{
+			name: "Babe Start Error",
+			fields: fields{
+				mockNetworkAPI,
+				mockErrorBlockProducerAPI,
+			},
+			args: args{
+				req: &[]string{"babe", "start"},
+			},
+			exp:    "babe service started",
+			expErr: errors.New("babe resume error"),
+		},
+		{
+			name: "Babe Start OK",
+			fields: fields{
+				mockNetworkAPI,
+				mockBlockProducerAPI,
+			},
+			args: args{
+				req: &[]string{"babe", "start"},
+			},
+			exp: "babe service started",
+		},
+		{
+			name: "Network Stop Error",
+			fields: fields{
+				mockErrorNetworkAPI,
+				mockBlockProducerAPI,
+			},
+			args: args{
+				req: &[]string{"network", "stop"},
+			},
+			exp:    "network service stopped",
+			expErr: errors.New("network stop error"),
+		},
+		{
+			name: "Network Stop OK",
+			fields: fields{
+				mockNetworkAPI,
+				mockBlockProducerAPI,
+			},
+			args: args{
+				req: &[]string{"network", "stop"},
+			},
+			exp: "network service stopped",
+		},
+		{
+			name: "Network Start Error",
+			fields: fields{
+				mockErrorNetworkAPI,
+				mockBlockProducerAPI,
+			},
+			args: args{
+				req: &[]string{"network", "start"},
+			},
+			exp:    "network service started",
+			expErr: errors.New("network start error"),
+		},
+		{
+			name: "Network Start OK",
+			fields: fields{
+				mockNetworkAPI,
+				mockBlockProducerAPI,
+			},
+			args: args{
+				req: &[]string{"network", "start"},
+			},
+			exp: "network service started",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			m := &DevModule{
+				networkAPI:       tt.fields.networkAPI,
+				blockProducerAPI: tt.fields.blockProducerAPI,
+			}
+			res := ""
```

Comment:
supernit usually use `var xxx type` to initialize a variable to its zero value. More of a convention, and really supernit

```suggestion
			var res string
```